RELEASE_NOTES.md

```diff
@@ -1,5 +1,6 @@
 # Unreleased
 
+- Remove support for slf4j 1.x, which is no longer maintained (#2902)
```

Comment:
This is ok, but I would have liked something like "Starting with this release, Jdbi uses features from slf4j 2.x and some parts will no longer work with version 1.x. If you are still using Jdbi with slf4j, you are strongly recommended to upgrade"

We are not dropping support because 1.7.x is no longer maintained, we are dropping support because we want to use newer 2.x features.